### PR TITLE
Fix `entryLocale` validation code

### PR DESCRIPTION
**Pull request by ids1024**

### CODE REVIEW

**Comment by ids1024** [comment]
Previously, `linuxdeploy` raised an error if the desktop file contained a line with a `-` in the key and a locale, since the locale checking code didn't allow `-`, and was applied to `entryName`. This seems to have broken AppImage builds for [Popsicle](https://github.com/pop-os/popsicle/).

**Comment by ids1024** [comment]
Oh, I see @mmstick opened an issue for this.

This should fix https://github.com/linuxdeploy/linuxdeploy-desktopfile/issues/5.

File: src/desktopfilereader.cpp [hunk 1]
```diff
@@ -137,11 +137,12 @@ namespace linuxdeploy {
...
 
                                     // locale may only contain A-Za-z- characters according to specification
-                                    for (const char c : entryName) {
+                                    for (const char c : entryLocale) {
                                         if (!(
                                             (c >= 'A' && c <= 'Z') ||
                                             (c >= 'a' && c <= 'z') ||
-                                            (c == '_') || (c == '@')
+                                            (c == '_') || (c == '@') ||
+                                            (c == '[') || (c == ']')
```
Comment (TheAssassin) [inline src/desktopfilereader.cpp]: This permits the use of `[` characters inside the locale part. I think you should remove this line, and iterate over the inner part only.

**Review by TheAssassin — CHANGES_REQUESTED** [review]

File: src/desktopfilereader.cpp (hunk 1 above)
Comment (TheAssassin) [reply src/desktopfilereader.cpp]: Actually, no, this is handled elsewhere already. However, I find it confusing to permit these here. I am going to update your branch to iterate only over the inner part of the string.

**Comment by TheAssassin** [comment]
We need to add a test case to make sure this won't happen again in the future.

**Comment by TheAssassin** [comment]
Seems like the existing check is already pretty broken. We have a test that is failing already on a locale like `de_DE.UTF-8` (which, according to https://specifications.freedesktop.org/desktop-entry-spec/latest/ar01s05.html, should be valid). I'll fix that as well.

**Review by TheAssassin — APPROVED** [review]

**Pull request merged**